### DISCUSSION TITLE
Contribution UI / ideas

### DISCUSSION THREAD

**Discussion by h16nning** [h16nning/skola#30] [General]
Hey @eMerzh!

You mentioned in #29 you would like to contribute to UI as well.

I don't have that many concrete plans, so you can't really disrupt them 🥴 Is there any specific feature or thing you would want to work on or something you want to have?

Maybe we can discuss that here...

**Comment by h16nning** [comment]
I started writing down some of my general ideas in issues as well

**Reply by eMerzh** [reply]
cool really nice idea :) i'll chime here and on issues as well ...

**Comment by eMerzh** [comment]
Sooo, i tested Mochi, and aside from payed, closed and few itches, it feels really nice to me... an it's some kind of inspiration...


# Decks in sidebar
To be more concrete... i would liked to have the decks in the sidebar in a tree like structure

<img width="237" alt="Capture d’écran 2024-02-07 à 20 14 42" src="https://github.com/h16nning/skola/assets/177003/11798f48-d00d-4675-8d11-cbff934aeae5">

This will ease the navigation, and add more visibility to subdecks. And maybe liberate a bit of space in the deck screen (where it feels a bit strange IMHO)


# Card List

I like to have a bit more details (overview) on the cards in a deck... maybe having a grid of the cards + decks might be interesting
(not sure i want all the config mochi has... but... being able to see them is nice)


# Card Creation
IMHO, one big pain to doing flashcard (ok i know it's part of the process) is to create the cards...
I saw somewhere (not sure where)  a big Textarea like... and you  typed  ">>" it auto flipped to the back card ... found it nice :) 
you could even do a copy pasta, and create multiple cards.

Also, we could probably simplify the toolbar to look more clean...(bold & friends only on selection for example)

# card Review
I like the card presentation, when the card (represented as a card) doesn't take your whole screen if you want to.

<img width="591" alt="Capture d’écran 2024-02-07 à 20 32 13" src="https://github.com/h16nning/skola/assets/177003/d2f03f38-8e85-472d-8463-198ac80204c5">

I also like the small tips about keyboard nav that helps to be very efficient.

<img width="507" alt="Capture d’écran 2024-02-07 à 20 30 47" src="https://github.com/h16nning/skola/assets/177003/d965c495-89c2-4c53-b89b-160ebd77a035">

I would reduce the importance of the progress bar and a number of item vs number in deck.

Also, i would offer the possibility to continue craming cards...
Like the way I used it (especially with kids) is that they need to learn something, and we want to start doing a few time in the deck then only after we might use the "spaced" (as in days) repetition.

Anyway... that was a few idea i had... some easier than other... 
i'm happy to discuss it ofc

**Comment by h16nning** [comment]
Hey, thank you for your detailed response! I agree with your observations.

### Sidebar
Offering a collapsible view of all decks is a good idea. I like the idea of placing it in the sidebar which makes it always quickly accessible. This would be especially helpful on desktop, less on mobile I think.

### Browse Decks
On mobile, I would also be useful to have a whole page of all your decks. Anki has this on the "start screen", which I don't like so much. In Anki (for university) I have dozens of decks and thousands of new cards (my daily limit is infinite) and those to review. Being greeted by hundreds of numbers and rows of decks is a really overwhelming feeling imo. Having this as a separate page "browse decks" or something like that would be ideal I think. Possible we could reuse the same component from the sidebar.

### Today View
Instead I was planing on having the today view (#36) as the starting point. This should serve as a sort of feed showing you only the relevant decks with due cards or new cards to do (maybe also showing an estimated time needed to review each deck).

I would ditch the now HomeView altogether or make it to browse decks.

Of course we could have a setting for switching the preferred starting page.

### Subdecks
Showing subdecks inside a deck strange right now, I agree. Especially how it is sitting below the card section. It also makes navigating to lower-order decks very tedious.

> The reason why I integrated it is because I wanted to have a focussed view of all decks within a deck. Let me explain:
Say you use Skola to learn a new language e.g. Spanish, geography facts and to memorize peoples names. If I just wanted to do geography, I would open the (highest-order) geography deck and could hide everything Spanish and people-related, so that they don't occupy me unconsciously.
> That's the idea at least. Maybe this focussed deck view (which I personally miss in anki) is not as important as I think and rather a very specific feature. 😂

We could omit the sub decks for now and consider adding something like it later.

### Card List
The card list is an interesting feature. I don't know if I agree with Mochi's way to put them upfront. I want to avoid seeing to much of my cards before I actively decide to learn them.

I would like to keep the little box with the three key information and the learning button (HeroDeckSection) at the top.

Maybe we could have the card list below (where subdecks reside now). I would consider having only a preview of them from where you could tab into a whole separate card list view.

I just made a really dirty sketch with figma of how a deck view could look on mobile to visualize what I have said.
<img width="367" alt="Bildschirmfoto 2024-02-07 um 22 37 16" src="https://github.com/h16nning/skola/assets/48356881/007d12f7-7823-4848-a653-93dc47699313">

Another thing I really like about Mochi is the notebook feature, but with the cards always showing the answer. This is not necessarily for learning but rather reading through your card collection like with normal notes (e. g. to look something up). Not being able to do that is an issue I have with Anki.

### Card creation
Absolutely. Card creation is generally tedious and especially with Skola right now.

Adding hotkeys (ctrl+enter etc. would be very useful already). Hinting at them like Mochi does is even better.

The thing with ">>" sounds cool. I think mochi uses "---". We could also use something like it to change the card type, make cloze fields and such. I really like that.

Inline formatting is useful too. Shouldn't be too hard to do, the tiptap editor has already a component for that.
In settings there is already a setting for editing. We could allow users to decide whether they prefer a clean look with inline options or a toolbar.

### Reviewing
Adding shadows and a border (if you meant that by represented as a card) is something I also thought about doing. It makes the cards feel more real (and by that important 😂🤔).

We can reduce the progress bar size and adf more information on how many cards are left / in the deck.

#### Cramming
I don't know how you mean by cramming. There is already an option to review cards when there are not due (Practice anyways). This button can maybe get more sophisticated (option to choose which of the cards to do again etc.)

We also need a daily card limit like in Anki and an option to ignore it haha.


What do you think about this?

**Comment by h16nning** [comment]
> I saw somewhere (not sure where) a big Textarea like... and you typed ">>" it auto flipped to the back card ... found it nice :)

I searched a bit. Were you talking about RemNote?

**Reply by eMerzh** [reply]
corrrrrrect.... well done sir 👏  ....

not sure about the "create cards" in a bullet list, but yeah that's the idea...  :)

**Reply by eMerzh** [reply]
i think this would be plausible with tiptap probably
